subsys/net/ip/tcp.c

```diff
@@ -2430,19 +2430,40 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		} else if (th && FL(&fl, ==, ACK, th_seq(th) == conn->ack)) {
 			tcp_send_timer_cancel(conn);
 			next = TCP_FIN_WAIT_2;
+		} else if (th && (len > 0)) {
+				tcp_out(conn, FIN | ACK);
```

Comment:
This sends out a FIN ACK with the wrong sequence number, please take this line from my closed PR.

Reply:
It should be: `tcp_out_ext(conn, (FIN | ACK), NULL, conn->seq - 1);`

Reply:
@ssharks I totally forgot our fork was public LOL
Thanks a lot for dropping b :-) 

Ah yes, that's from your new PR.
This affects only our SEQ number, not the number we are ACK-ing, correct?

Reply:
It indeed only affects the SEQ number, which is important as otherwise the receiving side gets out of sync

Reply:
So the receiving side (not zephyr) then  ACK-s wrong sequence number?
But practically it's not relevant if we anyway sent out FIN?

Tests show that it still works, at least with the hosts I had, which might not be representable, of course